### PR TITLE
chore: EXC-1735: Apply priority credit at the round start

### PR DESCRIPTION
**Pull request by berestovskyy**
Since priority credit is now serialized in the state, applying it at the beginning of the round offers functional equivalence while simplifying the code.

### CODE REVIEW

**Review by alexandru-uta — COMMENTED** [review]
thanks, lgtm

**Review by pakhomov-dfinity — APPROVED** [review]

**Review by dsarlis — APPROVED** [review]

File: rs/replicated_state/src/canister_state.rs [hunk 1]
```diff
@@ -144,10 +144,14 @@ impl CanisterState {
...
 
-    /// Apply priority credit
+    /// Applies priority credit and resets long execution mode.
     pub fn apply_priority_credit(&mut self) {
         self.scheduler_state.accumulated_priority -=
             std::mem::take(&mut self.scheduler_state.priority_credit);
+        // Aborting a long-running execution moves the canister to the
+        // default execution mode because the canister does not have a
+        // pending execution anymore.
+        self.scheduler_state.long_execution_mode = LongExecutionMode::default();
```
Comment (alin-at-dfinity) [inline rs/replicated_state/src/canister_state.rs:154]: This is only called from one location, behind an `if !has_aborted_or_paused_execution` check. So resetting `long_execution_mode` is always a no-op. I would instead add a `debug_assert!()` that `long_execution_mode == LongExecutionMode::default()`.

File: rs/state_manager/src/tip.rs [hunk 2]
```diff
@@ -1003,8 +1003,6 @@ fn serialize_canister_to_tip(
             metrics,
         )?;
 
-    // Priority credit must be zero at this point
-    assert_eq!(canister_state.scheduler_state.priority_credit.get(), 0);
```
Comment (alin-at-dfinity) [inline rs/state_manager/src/tip.rs:1007]: This appears to have (partly) been a proxy for `canister_state.scheduler_state.long_execution_mode == LongExecutionMode::default()`. Could we add that check here? Or does it not always hold?

File: rs/execution_environment/src/scheduler.rs [hunk 3]
```diff
@@ -571,25 +571,16 @@ impl SchedulerImpl {
     /// and `GlobalTimer` tasks that are carried out prior to processing
     /// any input messages.
     /// It also returns the list of canisters that have non-zero priority credit.
```
Comment (alin-at-dfinity) [inline rs/execution_environment/src/scheduler.rs:573]: This doc comment needs updating.

**Review by alin-at-dfinity — APPROVED** [review]

File: rs/replicated_state/src/canister_state.rs (hunk 1 above)
Comment (berestovskyy) [reply rs/replicated_state/src/canister_state.rs:154]: It does not look like a no-op to me. We set in the `long_execution_mode` to `Prioritized` in [apply_scheduling_strategy](https://github.com/dfinity/ic/blob/90e12e0aae718198304e8ed42464171e5878183c/rs/execution_environment/src/scheduler.rs#L382), and we reset it here.

**Pull request merged**